examples/gno.land/r/demo/profile/profile.gno

```diff
@@ -0,0 +1,312 @@
+		}
+	} else if username != "" {
+		if addr, exists := usernameIndex.Get(username); exists {
+			if value, exists := profiles.Get(addr.(string)); exists {
+				return value.(UserProfile).Gender
+			}
+		}
+	}
+	return "Address or Username not found"
+}
+
+// Display website from address or username
+func DisplayWebsite(address std.Address, username string) string {
+	if address != "" {
+		if value, exists := profiles.Get(address.String()); exists {
+			return value.(UserProfile).Website
+		}
+	} else if username != "" {
+		if addr, exists := usernameIndex.Get(username); exists {
+			if value, exists := profiles.Get(addr.(string)); exists {
+				return value.(UserProfile).Website
+			}
+		}
+	}
+	return "Address or Username not found"
+}
+
+// Display country from address or username
+func DisplayCountry(address std.Address, username string) string {
+	if address != "" {
+		if value, exists := profiles.Get(address.String()); exists {
+			return value.(UserProfile).Country
+		}
+	} else if username != "" {
+		if addr, exists := usernameIndex.Get(username); exists {
+			if value, exists := profiles.Get(addr.(string)); exists {
+				return value.(UserProfile).Country
+			}
+		}
+	}
+	return "Address or Username not found"
+}
+
+// Display city from address or username
+func DisplayCity(address std.Address, username string) string {
+	if address != "" {
+		if value, exists := profiles.Get(address.String()); exists {
+			return value.(UserProfile).City
+		}
+	} else if username != "" {
+		if addr, exists := usernameIndex.Get(username); exists {
+			if value, exists := profiles.Get(addr.(string)); exists {
+				return value.(UserProfile).City
+			}
+		}
+	}
+	return "Address or Username not found"
+}
+
+// Display modification history specific to a field from address or username
+func DisplayModificationHistory(address std.Address, username, field string) string {
+	if address != "" {
+		if value, exists := profiles.Get(address.String()); exists {
+			if value.(UserProfile).ModificationHistory != nil {
+				if changes, exists := value.(UserProfile).ModificationHistory[field]; exists {
+					var b bytes.Buffer
+					for _, change := range changes {
+						b.WriteString(ufmt.Sprintf("%s\n", change))
+					}
+					return b.String()
+				}
+			}
+		}
+	} else if username != "" {
+		if addr, exists := usernameIndex.Get(username); exists {
+			if value, exists := profiles.Get(addr.(string)); exists {
+				if value.(UserProfile).ModificationHistory != nil {
+					if changes, exists := value.(UserProfile).ModificationHistory[field]; exists {
+						var b bytes.Buffer
+						for _, change := range changes {
+							b.WriteString(ufmt.Sprintf("%s\n", change))
+						}
+						return b.String()
+					}
+				}
+			}
+		}
+	}
+	return "Address or Username not found or no modifications recorded for the field"
+}
+
+func Render(path string) string {
+	if path == "" {
+		return renderHome()
+	}
+
+	return "unknown path"
+}
+
+func renderHome() string {
```

Comment:
Is there any way to hide your own information when the content is returned to another address? Will be all or nothing?

Reply:
Exactly. Despite you can search the other information in gnoscan, you should be able to hide it just in case you don't want to show something anymore.